cmd/agent/common/commands/check.go

```diff
@@ -163,8 +165,13 @@ func Check(loggerName config.LoggerName, confFilePath *string, flagNoColor *bool
 				discoveryRetryInterval = discoveryTimeout
 			}
 
-			allConfigs := common.WaitForConfigs(time.Duration(discoveryRetryInterval)*time.Second, time.Duration(discoveryTimeout)*time.Second,
-				common.SelectedCheckMatcherBuilder([]string{checkName}, discoveryMinInstances))
+			// Create the CheckScheduler, but do not attach it to
+			// AutoDiscovery.  NOTE: we do not start common.Coll, either.
+			collector.InitCheckScheduler(common.Coll)
+
+			waitForConfigsFromADCtx, cancelTimeout := context.WithTimeout(context.Background(), time.Duration(discoveryTimeout)*time.Second)
+			allConfigs := waitForConfigsFromAD(waitForConfigsFromADCtx, checkName, int(discoveryMinInstances))
+			cancelTimeout()
```

Comment:
Would it make sense to apply these changes to `common.WaitForConfigs` instead of doing them locally? The only other user for it is `agent jmx` command, which, AIUI, is very similar to `agent check` and would likely benefit from the fixes you are doing here.